# DSCI 631: House Price Prediction Project  <br> Term Project Phase 1: Scoping an analytics project

## I. Group members 
- Group member 
    - Name: Anh Nguyen
    - Email: an928@drexel.edu
- Group member 
    - Name: Thuy Hong Doan
    - Email: td688@drexel.edu
- Group member 
    - Name: Alex Hoang
    - Email: ah3754@drexel.edu

### Team members background:

- __Anh Nguyen__:  I have a BA in Mathematics and after graduation, I worked as a healthcare market research analyst. Specifically, I worked with insurance claim data to track and forecast demand for Medtech devices and medical procedures. I have worked with Python, SQL and C. I will leverage my previous professional experience to successfully work with this dataset and analyze important trends. 

- __Alex Hoang__: I have a BA in Management Information Systems and I used to work as a front-end developer. My professional coding knowledge will be helpful when handling and storing the data in the required format and to successfully write analytics programs.

- __Thuy Hong Doan__: graduated with a bachelor of business administration in Computer Information Systems. Her skill set varies from different programming languages to analytical and problem-solving skills. She worked with HTML, CSS, SQL, JavaScript, and Python. She plans to use her previous coding experience and understanding of data structure to help successfully complete this project.

#### Growth area :

...

## II. Topic Summary 
The dataset contains information on residential homes in Ames, Iowa, USA, including 79 features such as the size of the lot, the number of bedrooms and bathrooms, and various other features related to the property. The goal of this project is to develop a machine learning model that can accurately predict the sale price of a house based on these features. The dataset requires some data cleaning and preprocessing, such as filling in missing values and converting categorical variables to numerical ones. Additional feature engineering may also be required to extract more meaningful information from the data.

## III. Term Project Phase 1: details 

**1. Source of the data**

- This data comes from Kaggle's competition "House Prices - Advanced Regression Techniques". The goal of the competition is to predict sales price using the Ames Housing dataset compiled by Dean De Cock.
- Link: https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques
- The data are splitted into two seperate files, a test file and a train file. Each file has 1,460 values (rows) not including the label. Each file has 81 variables (columns).

**2. Definition of variables:**

- SalePrice - the property's sale price in dollars. This is the target variable that we are trying to predict.
- Id: ID of the property
- MSSubClass: The building class
- MSZoning: The general zoning classification
- LotFrontage: Linear feet of street connected to property
- LotArea: Lot size in square feet
- Street: Type of road access
- Alley: Type of alley access
- LotShape: General shape of property
- LandContour: Flatness of the property
- Utilities: Type of utilities available
- LotConfig: Lot configuration
- LandSlope: Slope of property
- Neighborhood: Physical locations within Ames city limits
- Condition1: Proximity to main road or railroad
- Condition2: Proximity to main road or railroad (if a second is present)
- BldgType: Type of dwelling
- HouseStyle: Style of dwelling
- OverallQual: Overall material and finish quality
- OverallCond: Overall condition rating
- YearBuilt: Original construction date
- YearRemodAdd: Remodel date
- RoofStyle: Type of roof
- RoofMatl: Roof material
- Exterior1st: Exterior covering on house
- Exterior2nd: Exterior covering on house (if more than one material)
- MasVnrType: Masonry veneer type
- MasVnrArea: Masonry veneer area in square feet
- ExterQual: Exterior material quality
- ExterCond: Present condition of the material on the exterior
- Foundation: Type of foundation
- BsmtQual: Height of the basement
- BsmtCond: General condition of the basement
- BsmtExposure: Walkout or garden level basement walls
- BsmtFinType1: Quality of basement finished area
- BsmtFinSF1: Type 1 finished square feet
- BsmtFinType2: Quality of second finished area (if present)
- BsmtFinSF2: Type 2 finished square feet
- BsmtUnfSF: Unfinished square feet of basement area
- TotalBsmtSF: Total square feet of basement area
- Heating: Type of heating
- HeatingQC: Heating quality and condition
- CentralAir: Central air conditioning
- Electrical: Electrical system
- 1stFlrSF: First Floor square feet
- 2ndFlrSF: Second floor square feet
- LowQualFinSF: Low quality finished square feet (all floors)
- GrLivArea: Above grade (ground) living area square feet
- BsmtFullBath: Basement full bathrooms
- BsmtHalfBath: Basement half bathrooms
- FullBath: Full bathrooms above grade
- HalfBath: Half baths above grade
- Bedroom: Number of bedrooms above basement level
- Kitchen: Number of kitchens
- KitchenQual: Kitchen quality
- TotRmsAbvGrd: Total rooms above grade (does not include bathrooms)
- Functional: Home functionality rating
- Fireplaces: Number of fireplaces
- FireplaceQu: Fireplace quality
- GarageType: Garage location
- GarageYrBlt: Year garage was built
- GarageFinish: Interior finish of the garage
- GarageCars: Size of garage in car capacity
- GarageArea: Size of garage in square feet
- GarageQual: Garage quality
- GarageCond: Garage condition
- PavedDrive: Paved driveway
- WoodDeckSF: Wood deck area in square feet
- OpenPorchSF: Open porch area in square feet
- EnclosedPorch: Enclosed porch area in square feet
- 3SsnPorch: Three season porch area in square feet
- ScreenPorch: Screen porch area in square feet
- PoolArea: Pool area in square feet
- PoolQC: Pool quality
- Fence: Fence quality
- MiscFeature: Miscellaneous feature not covered in other categories
- MiscVal: $Value of miscellaneous feature
- MoSold: Month Sold
- YrSold: Year Sold
- SaleType: Type of sale
- SaleCondition: Condition of sale

**3. What is the availability of relevant pre-processed data?**


**4. What will the analysis do, and who/what it will support?** 



**5. Who would be interested in the data analysis?**

- 
- 
- 
**6. How your analysis might be completed and disseminated?**


**7. How your analysis might fit into an application or investigation?**

 


**8. Limitations**

- 
- 
- 

**9. Potential solutions** 

- 
- 
- 

### Potential Software used 
- Juypter notebook 

### Hardware used 
- 
- 
- Macbook Pro processor i5 RAM 8GB

## IV. Exploratory Data Analysis 


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("./house-prices-advanced-regression-techniques/train.csv")

In [3]:
df.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

In [7]:
# Check for number of null values
for col in df.columns:
    if df[col].isnull().sum() != 0:
        print(col, ":", df[col].isnull().sum())

LotFrontage : 259
Alley : 1369
MasVnrType : 8
MasVnrArea : 8
BsmtQual : 37
BsmtCond : 37
BsmtExposure : 38
BsmtFinType1 : 37
BsmtFinType2 : 38
Electrical : 1
FireplaceQu : 690
GarageType : 81
GarageYrBlt : 81
GarageFinish : 81
GarageQual : 81
GarageCond : 81
PoolQC : 1453
Fence : 1179
MiscFeature : 1406


In [8]:
# Check for number of unique values
for col in df.columns:
    # get the number of unique values in the column
    num_unique_values = df[col].nunique()
    # print the result
    print(col, ":", num_unique_values)

Id : 1460
MSSubClass : 15
MSZoning : 5
LotFrontage : 110
LotArea : 1073
Street : 2
Alley : 2
LotShape : 4
LandContour : 4
Utilities : 2
LotConfig : 5
LandSlope : 3
Neighborhood : 25
Condition1 : 9
Condition2 : 8
BldgType : 5
HouseStyle : 8
OverallQual : 10
OverallCond : 9
YearBuilt : 112
YearRemodAdd : 61
RoofStyle : 6
RoofMatl : 8
Exterior1st : 15
Exterior2nd : 16
MasVnrType : 4
MasVnrArea : 327
ExterQual : 4
ExterCond : 5
Foundation : 6
BsmtQual : 4
BsmtCond : 4
BsmtExposure : 4
BsmtFinType1 : 6
BsmtFinSF1 : 637
BsmtFinType2 : 6
BsmtFinSF2 : 144
BsmtUnfSF : 780
TotalBsmtSF : 721
Heating : 6
HeatingQC : 5
CentralAir : 2
Electrical : 5
1stFlrSF : 753
2ndFlrSF : 417
LowQualFinSF : 24
GrLivArea : 861
BsmtFullBath : 4
BsmtHalfBath : 3
FullBath : 4
HalfBath : 3
BedroomAbvGr : 8
KitchenAbvGr : 4
KitchenQual : 4
TotRmsAbvGrd : 12
Functional : 7
Fireplaces : 4
FireplaceQu : 5
GarageType : 6
GarageYrBlt : 97
GarageFinish : 3
GarageCars : 5
GarageArea : 441
GarageQual : 5
GarageCond : 5
PavedDr

**Continuous variables and numbers of unique values**

- LotFrontage : 110
- LotArea : 1073
- OverallQual : 10
- OverallCond : 9
- YearBuilt : 112
- YearRemodAdd : 61
- MasVnrArea : 327
- BsmtFinSF1 : 637
- BsmtFinSF2 : 144
- BsmtUnfSF : 780
- TotalBsmtSF : 721
- 1stFlrSF : 753
- 2ndFlrSF : 417
- LowQualFinSF : 24
- GrLivArea : 861
- BsmtFullBath : 4
- BsmtHalfBath : 3
- FullBath : 4
- HalfBath : 3
- BedroomAbvGr : 8
- KitchenAbvGr : 4
- TotRmsAbvGrd : 12
- Fireplaces : 4
- GarageYrBlt : 97
- GarageCars : 5
- GarageArea : 441
- WoodDeckSF : 274
- OpenPorchSF : 202
- EnclosedPorch : 120
- 3SsnPorch : 20
- ScreenPorch : 76
- PoolArea : 8
- MiscVal : 21
- MoSold : 12
- YrSold : 5
- SalePrice : 663

**Categorical variables of numbers of unique values**

- Id : 1460
- MSSubClass : 15
- MSZoning : 5
- Street : 2
- Alley : 2
- LotShape : 4
- LandContour : 4
- Utilities : 2
- LotConfig : 5
- LandSlope : 3
- Neighborhood : 25
- Condition1 : 9
- Condition2 : 8
- BldgType : 5
- HouseStyle : 8
- RoofStyle : 6
- RoofMatl : 8
- Exterior1st : 15
- Exterior2nd : 16
- MasVnrType : 4
- ExterQual : 4
- ExterCond : 5
- Foundation : 6
- BsmtQual : 4
- BsmtCond : 4
- BsmtExposure : 4
- BsmtFinType1 : 6
- BsmtFinType2 : 6
- Heating : 6
- HeatingQC : 5
- CentralAir : 2
- Electrical : 5
- KitchenQual : 4
- Functional : 7
- FireplaceQu : 5
- GarageType : 6
- GarageFinish : 3
- GarageQual : 5
- GarageCond : 5
- PavedDrive : 3
- PoolQC : 3
- Fence : 4
- MiscFeature : 4
- SaleType : 9
- SaleCondition : 6